src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -253,7 +253,9 @@ struct nonmaxsuppression
                                return std::make_pair(sc, box_idx - 1);
                            });
         }
-        std::sort(std::execution::par, boxes_heap.begin(), boxes_heap.end());
+        std::sort(std::execution::par, boxes_heap.begin(), boxes_heap.end(), [&](auto a, auto b) {
```

Comment:
Use `std::greater<>{}` instead.

Reply:
std::pair uses lexicographical comparison, which means it would only compare the second value if the first value is equal.

If two scores are equal the sorting order is ambiguous(unless you use `stable_sort`), so comparing the second value is not a problem, but it does make the sorting more consistent without using a slower sort algo. This also matches the previous behavior with `priority_queue`.

Reply:
Yeah I saw that about pair comparison, but I guess that makes sense if we wanted to keep it consistent with the previous priority_queue  implementation. I'll update

Reply:
done